src/charm.py

```diff
@@ -721,6 +764,9 @@ def _handle_ingress_data_provided(self, event: RelationEvent):
         # update-status.
         self._process_status_and_configurations()
 
+        if self.forward_auth.is_ready():
```

Comment:
Does it matter if the `forward_auth` provider side is ready? I think that we only need to check if `enable_experimental_forward_auth` is true.